# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/html/HTMLListAccessible.cpp
# Commit: 700fc3dd5142
# Full Hash: 700fc3dd5142f692c634e384eb93608a81ca024c
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-06-10 03:46:06
# Description:
#   Bug 1715151 - Check if we have a frame for HTMLLIAccessible when getting bounds. r=morgan
#   
#   I'm not sure in what markup/style scenario a list item accessible
#   wouldn't have a frame, but would have a bullet. Apparently that can
#   happen in the wild. Added an assertion, hopefully we can trip it on
# ==============================================================================

diff -r 5e0ac6d5d01e -r 700fc3dd5142 accessible/html/HTMLListAccessible.cpp
--- a/accessible/html/HTMLListAccessible.cpp	Wed Jun 09 19:27:07 2021 +0000
+++ b/accessible/html/HTMLListAccessible.cpp	Wed Jun 09 19:33:26 2021 +0000
@@ -55,8 +55,12 @@
   nsRect rect = AccessibleWrap::BoundsInAppUnits();
 
   LocalAccessible* bullet = Bullet();
-  if (bullet && GetFrame()->StyleList()->mListStylePosition !=
-                    NS_STYLE_LIST_STYLE_POSITION_INSIDE) {
+  nsIFrame* frame = GetFrame();
+  MOZ_ASSERT(!(bullet && !frame), "Cannot have a bullet if there is no frame");
+
+  if (bullet && frame &&
+      frame->StyleList()->mListStylePosition !=
+          NS_STYLE_LIST_STYLE_POSITION_INSIDE) {
     nsRect bulletRect = bullet->BoundsInAppUnits();
     return rect.Union(bulletRect);
   }
